app/database/server/models/custom_emoji.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Model} from '@nozbe/watermelondb';
+import {field} from '@nozbe/watermelondb/decorators';
+
+import {MM_TABLES} from '@constants/database';
+
+const {CUSTOM_EMOJI} = MM_TABLES.SERVER;
+
+/**  The CustomEmoji model describes all the custom emojis used in the Mattermost app */
```

Comment:
```suggestion
/**  The CustomEmoji model describes the custom emojis availabe in the Mattermost server */
```

Reply:
Are we not going to overwrite the raw id for this record at the time of creation? Or does 'Id' stands for another column name for this entity?

Reply:
I'm afraid I don't follow your question.

The Model is one thing and YES we will be overriding the value of the id on creation, but the model **type definition** as in the .d.ts file should have the **id** defined, so whenver we do something like
```ts
const result: CustomEmoji = Query;
console.log(reault.id);
```
Typescript rus smooth as the id is defined, yes?

Reply:
Ooh I see.  This has been missed out from all the other models' definition as well.  I will get it :)

Reply:
![TS - Model Id](https://user-images.githubusercontent.com/43164508/103574835-149bca00-4eea-11eb-8a7c-f18b6821da07.png)


Hey @enahum , I cross-checked the 'id' on the DB Manager branch and Typescript does run smooth with how the models are currently defined.  In the screen-shot above, I was hovering over the deconstructed 'id' field. 

TS is not complaining as the 'TS Definition' for the 'Servers' class for example, extends the Model TS definition in which the 'id' field is defined as a RecordId.

`export default class Servers extends Model {`

Hence, no need to add the 'id' field to each model again.